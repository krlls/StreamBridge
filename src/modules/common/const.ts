export const limits = {
  music: {
    pagination: {
      maxLimit: 50,
      maxOffset: 60,
    },
  },
}
