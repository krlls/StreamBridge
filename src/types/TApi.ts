export namespace Api {
  export namespace User {
    export const PREFIX = '/user'

    export namespace Create {
      export const URL = '/create'

      export type Req = {
        login: string,
        pass: string,
      }
    }
  }
}
