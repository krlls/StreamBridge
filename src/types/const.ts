const TYPES = {
  UserController: Symbol.for('UserController'),

  UserService: Symbol.for('UserService'),

  UserRepository: Symbol.for('UserRepository'),

  UserEntityConverter: Symbol.for('UserEntityConverter'),
}

export { TYPES }
