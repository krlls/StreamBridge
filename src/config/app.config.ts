import { ServerConfig } from '../types/TServer'
import { E_NODE_ENV } from '../types/common'
import { NODE_ENV } from './env'

export const serverConfig: ServerConfig = {
  port: 3000,
  silent: NODE_ENV === E_NODE_ENV.TEST,
}
