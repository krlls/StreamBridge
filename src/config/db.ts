import { DataSourceOptions } from 'typeorm/data-source/DataSourceOptions'

import { E_NODE_ENV } from '../types/common'

const entitiesPatch = 'src/infra/db/Sqlite/entities/**/*.ts'
const dbPatch = 'data/db.sqlite3'
export const sqlLite: Record<E_NODE_ENV, DataSourceOptions> = {
  [E_NODE_ENV.TEST]: {
    type: 'better-sqlite3',
    database: ':memory:',
    entities: [entitiesPatch],
    synchronize: true,
  },
  [E_NODE_ENV.DEV]: {
    type: 'better-sqlite3',
    database: dbPatch,
    entities: [entitiesPatch],
    synchronize: true,
    migrationsRun: false,
    logging: true,
    enableWAL: true,
  },
  [E_NODE_ENV.PROD]: {
    type: 'better-sqlite3',
    database: dbPatch,
    entities: [entitiesPatch],
    synchronize: false,
    migrationsRun: true,
    logging: false,
    enableWAL: true,
  },
}
