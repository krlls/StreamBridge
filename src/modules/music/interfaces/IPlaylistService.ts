import { CreatePlaylistDTO } from '../dtos/CreatePlaylistDTO'
import { Playlist } from '../entities/Playlist'
import { ServiceResultDTO } from '../../../types/common'

export interface IPlaylistService {
  createPlayList(playlist: CreatePlaylistDTO): Promise<ServiceResultDTO<Playlist>>,
  getPlaylistById(id: number): Promise<ServiceResultDTO<Playlist | null>>,
  getPlaylistByExternalId(extenalId: string): Promise<ServiceResultDTO<Playlist | null>>,
  getUserPlaylists(userId: number): Promise<ServiceResultDTO<Playlist[]>>,
}
