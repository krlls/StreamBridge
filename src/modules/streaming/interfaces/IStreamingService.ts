import { ServiceResultDTO } from '../../../types/common'
import { Streaming } from '../entities/Streaming'
import { CreateStreamingDTO } from '../dtos/CreateStreamingDTO'
import { LoginUrlDTO } from '../dtos/LoginUrlDTO'
import { CreateLoginUrlDTO } from '../dtos/createLoginUrlDTO'

export interface IStreamingService {
  createStreaming(steamingData: CreateStreamingDTO): Promise<ServiceResultDTO<Streaming>>,
  getLoginUrl(data: CreateLoginUrlDTO): Promise<ServiceResultDTO<LoginUrlDTO>>,
  // getStreaming(userId: number, streamingType: string): Promise<ServiceResultDTO<Streaming>>,
  // getSreamingById(steamingId: number): Promise<ServiceResultDTO<Streaming>>,
  // getSreamingsByUserId(userId: number): Promise<ServiceResultDTO<Streaming[]>>,
}
