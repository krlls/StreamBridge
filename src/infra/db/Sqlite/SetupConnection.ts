import { DataSource, EntityTarget, Repository } from 'typeorm'
import Database from 'better-sqlite3'

import { NODE_ENV, sqlLite } from '../../../config'

export const dataSource = new DataSource(sqlLite[NODE_ENV])

export class SqliteDB {
  private static _instance: SqliteDB

  public static get instance(): SqliteDB {
    if (!this._instance) this._instance = new SqliteDB()

    return this._instance
  }

  private dbConnect!: DataSource
  private testdb!: any

  async setupTestDB() {
    this.testdb = new Database('')
    this.dbConnect = await dataSource.initialize()
  }

  async teardownTestDB() {
    await this.dbConnect.destroy()
    this.testdb.close()
  }
}

export const getRepository = <Entity extends Record<string, any>>(target: EntityTarget<Entity>): Repository<Entity> => {
  return dataSource.getRepository(target)
}

export default dataSource
