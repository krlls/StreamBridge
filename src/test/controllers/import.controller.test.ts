import { expect, test, describe, beforeEach, afterEach } from '@jest/globals'

import { Api } from '../../types/TApi'
import { TestApp } from '../index.test'
import { SqliteDB } from '../../infra/db/Sqlite/SetupConnection'
import { authUrl, importUrl, testStreamingDTO, testUserData, userUrl } from '../helpers/test.helpers'
import { controllerContainer } from '../../inversify.config'
import { IStreamingService } from '../../modules/streaming/interfaces/IStreamingService'
import { TYPES } from '../../types/const'
import { IPlaylistService } from '../../modules/music/interfaces/IPlaylistService'

const streamingService = controllerContainer.get<IStreamingService>(TYPES.StreamingService)
const playlistsServise = controllerContainer.get<IPlaylistService>(TYPES.PlaylistService)
describe('Import media controller tests', () => {
  let testToken: string = ''
  let currentUser = {} as any

  beforeEach(async () => {
    await SqliteDB.instance.setupTestDB()

    const userResp = await TestApp.post(userUrl(Api.User.Create.URL)).send(testUserData)
    currentUser = JSON.parse(userResp.text)

    const resp = await TestApp.post(authUrl(Api.Auth.Login.URL)).send({
      login: testUserData.login,
      pass: testUserData.pass,
    })
    testToken = JSON.parse(resp.text).token
  })

  afterEach(async () => {
    await SqliteDB.instance.teardownTestDB()
  })

  test('Import playlists works', async () => {
    const streamingDTO = testStreamingDTO(currentUser.id)

    await streamingService.createStreaming(streamingDTO)

    const response = await TestApp.post(importUrl(Api.Import.Playlists.URL))
      .set({
        Authorization: `Bearer ${testToken}`,
      })
      .send({
        streamingType: Api.Streaming.EApiStreamingType.SPOTIFY,
      })

    expect(response.status).toBe(200)
    expect(response.text).toMatchSnapshot()
  })

  test('Import playlists without streaming does not work', async () => {
    const response = await TestApp.post(importUrl(Api.Import.Playlists.URL))
      .set({
        Authorization: `Bearer ${testToken}`,
      })
      .send({
        streamingType: Api.Streaming.EApiStreamingType.SPOTIFY,
      })

    expect(response.status).toBe(400)
  })

  test('Import tracks by playlist works', async () => {
    const streamingDTO = testStreamingDTO(currentUser.id)
    await streamingService.createStreaming(streamingDTO)

    await TestApp.post(importUrl(Api.Import.Playlists.URL))
      .set({
        Authorization: `Bearer ${testToken}`,
      })
      .send({
        streamingType: Api.Streaming.EApiStreamingType.SPOTIFY,
      })

    const playlists = (await playlistsServise.getUserPlaylists(currentUser.id)) as any[]

    const playlistsResponse = await TestApp.post(importUrl(Api.Import.Tracks.URL))
      .set({
        Authorization: `Bearer ${testToken}`,
      })
      .send({
        playlistId: playlists[0].id,
      })

    expect(playlistsResponse.status).toBe(200)
    expect(playlistsResponse.text).toMatchSnapshot()
  })
})
