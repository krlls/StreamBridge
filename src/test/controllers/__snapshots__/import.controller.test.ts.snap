// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Import media controller tests Import playlists works 1`] = `"{"exported":10,"saved":10}"`;

exports[`Import media controller tests Import tracks by playlist works 1`] = `"{"exported":150,"saved":150}"`;
